Applications/FileIO/GocadIO/GocadAsciiReader.cpp

```diff
@@ -397,14 +485,68 @@ bool GocadTSurfaceReader::parseNodes(
     return false;
 }
 
-bool GocadTSurfaceReader::parseElements(
+bool GocadAsciiReader::parseLineSegments(
+    std::ifstream& in,
+    std::vector<MeshLib::Node*>& nodes,
+    std::vector<MeshLib::Element*>& elems,
+    std::map<std::size_t, std::size_t> const& node_id_map,
+    MeshLib::Properties& mesh_prop)
+{
+    std::array<std::size_t, 3> data;
+    MeshLib::PropertyVector<int>& mat_ids =
+        *mesh_prop.getPropertyVector<int>(mat_id_name);
+    int current_mat_id(0);
+    if (!mat_ids.empty())
+    {
+        current_mat_id = (*std::max_element(mat_ids.begin(), mat_ids.end()))++;
+    }
+    std::streampos pos = in.tellg();
+    std::size_t id(0);
+    std::string line;
+    while (std::getline(in, line))
+    {
+        if (line.empty() || isCommentLine(line))
+        {
+            continue;
+        }
+        if (line.substr(0, 3) == "SEG")
+        {
+            std::stringstream sstr(line);
+            std::string keyword;
+            sstr >> keyword >> data[0] >> data[1];
```

Comment:
Also not clear is why `data` is 3 long, but only two elements are used.

Reply:
:heavy_check_mark: